common/tools/dev-tool/src/commands/run/bundle.ts

```diff
@@ -112,14 +123,28 @@ export default leafCommand(commandInfo, async (options) => {
           preferBuiltins: false,
           browser: true,
         }),
-        ...(options["polyfill-node"] ? [nodePolyfills({ sourceMap: true })] : []),
         cjs({
           dynamicRequireTargets: [globFromStore("chai")],
         }),
+
+        ...(options["inject-node-polyfills"]
+          ? [
+              inject({
+                modules: {
+                  Buffer: ["buffer", "Buffer"],
+                  Stream: ["stream", "Stream"],
+                  process: "process",
+                },
+              }),
+            ]
+          : []),
+        ...(options["polyfill-node"] ? [nodePolyfills({ sourceMap: true })] : []),
```

Comment:
what happens if you say yes polyfill node but you don't inject node polyfills? Should we throw an error or something? I guess I'm not sure what the other state means either.

Reply:
`polyfill-node` is an automatic solution that takes care of polyfilling end-to-end. `inject-node-polyfills` on the other hand just takes care of injecting the relevant import statements with the assumption that the polyfill library is already a dependency and will be bundled. It doesn't make sense to enable both at the same time but since one of them defaults to disabled, I am not worried too much about it. If you prefer, I can add a pass of option validation to weed such scenarios out.

Reply:
Minor preference to error if someone mistakenly leaves both enabled, but not blocking if you want to track it out of band

Reply:
Option validation is added in https://github.com/Azure/azure-sdk-for-js/pull/27305/commits/0623d932b1547a288f3651931bc7769e9e10d097.